### ISSUE TITLE
deploy_cluster.yml fails with branch release-3.10 on RHEL 7.5

### ISSUE THREAD

**Issue by adelton** [openshift/openshift-ansible#9755]
#### Description

On an all-on-one installation attempt `ansible-playbook openshift-ansible/playbooks/deploy_cluster.yml` fails while installing `atomic`.

##### Version

Please put the following version information in the code block
indicated below.

* Your ansible version per `ansible --version`

```
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
```

* The output of `git describe`

```
openshift-ansible-3.10.35-1-11-ga8d56b0
```

If you're running from playbooks installed via RPM

```
package openshift-ansible is not installed
```

##### Steps To Reproduce
1. `ansible-playbook -i ansible-inventory openshift-ansible/playbooks/prerequisites.yml`
1. `ansible-playbook -i ansible-inventory openshift-ansible/playbooks/deploy_cluster.yml`

##### Expected Results

Commands pass without failure.

##### Observed Results

The `prerequisites.yml passes`, the second command fails with

```
  1. Hosts:    master.example.com
     Play:     node bootstrap config
     Task:     install needed rpm(s)
     Message:  All items completed
```

and output before that (`\\n`s converted to newlines for readability) had

```
Error: Package: 1:atomic-1.22.1-22.git5a342e3.el7.x86_64 (rhel-7-server-extras-rpms)
           Requires: python-docker-py >= 1.7.2-1
           Available: docker-python-0.7.1-37.el7.x86_64 (rhel-7-server-extras-rpms)
               python-docker-py = 0.7.1-37.el7
           Available: docker-python-1.0.0-21.el7.x86_64 (rhel-7-server-extras-rpms)
               python-docker-py = 1.0.0-21
           Available: docker-python-1.0.0-22.el7.x86_64 (rhel-7-server-extras-rpms)
               python-docker-py = 1.0.0-22
           Available: docker-python-1.0.0-35.el7.x86_64 (rhel-7-server-extras-rpms)
               python-docker-py = 1.0.0-35
           Available: docker-python-1.0.0-53.el7.x86_64 (rhel-7-server-extras-rpms)
               python-docker-py = 1.0.0-53
           Available: docker-python-1.4.0-108.el7.x86_64 (rhel-7-server-extras-rpms)
               python-docker-py = 1.4.0-108.el7
           Available: docker-python-1.4.0-115.el7.x86_64 (rhel-7-server-extras-rpms)
               python-docker-py = 1.4.0-115.el7
           Available: python-docker-py-1.4.0-117.el7.noarch (rhel-7-server-extras-rpms)
               python-docker-py = 1.4.0-117.el7
           Available: python-docker-py-1.4.0-118.el7.noarch (rhel-7-server-extras-rpms)
               python-docker-py = 1.4.0-118.el7
           Available: python-docker-py-1.7.2-1.el7.noarch (rhel-7-server-extras-rpms)
               python-docker-py = 1.7.2-1.el7
           Available: python-docker-py-1.9.0-1.el7.noarch (rhel-7-server-extras-rpms)
               python-docker-py = 1.9.0-1.el7
           Available: python-docker-py-1.10.6-1.el7.noarch (rhel-7-server-extras-rpms)
               python-docker-py = 1.10.6-1.el7
           Available: python-docker-py-1.10.6-3.el7.noarch (rhel-7-server-extras-rpms)
               python-docker-py = 1.10.6-3.el7
           Available: python-docker-py-1.10.6-4.el7.noarch (rhel-7-server-extras-rpms)
               python-docker-py = 1.10.6-4.el7
```

##### Additional Information

Provide any additional information which may help us diagnose the
issue.

* Your operating system and version, ie: RHEL 7.2, Fedora 23 (`$ cat /etc/redhat-release`)

```
Red Hat Enterprise Linux Server release 7.5 (Maipo)
```

* Your inventory file (especially any non-standard configuration parameters)

```
[OSEv3:children]
masters
nodes

[OSEv3:vars]
ansible_ssh_user=root
deployment_type=origin
openshift_disable_check=memory_availability,docker_storage,disk_availability,package_version
openshift_repos_enable_testing=true
openshift_enable_excluders=false

osm_cluster_network_cidr=10.128.0.0/14
openshift_portal_net=172.30.0.0/16
osm_host_subnet_length=9

openshift_router_selector='node-router=true'
openshift_registry_selector='node-registry=true'
template_service_broker_selector={"node-app": "true"}
openshift_web_console_nodeselector={"node-app": "true"}

openshift_node_groups=[	{ 'name': 'node-all-on-one', 'labels': [ 'node-router=true', 'node-registry=true', 'node-app=true', 'node-role.kubernetes.io/master=true', 'node-role.kubernetes.io/infra=true', 'node-role.kubernetes.io/compute=true' ] }, { 'name': 'node-router', 'labels': [ 'node-router=true', 'node-role.kubernetes.io/infra=true' ] }, { 'name': 'node-registry', 'labels': [ 'node-registry=true', 'node-role.kubernetes.io/infra=true' ] }, { 'name': 'node-app', 'labels': [ 'node-app=true', 'node-role.kubernetes.io/compute=true' ] }, { 'name': 'node-master', 'labels': [ 'node-role.kubernetes.io/master=true' ] } ]

openshift_master_identity_providers=[{'name': 'allow_all', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]

[masters]
master.example.com

[etcd]
master.example.com

[nodes]
master.example.com openshift_schedulable=true openshift_node_labels="{ 'node-router': 'true', 'node-registry': 'true', 'node-app': 'true', 'node-master': 'true' }" openshift_node_group_name='node-all-on-one'
```

Unlike in https://github.com/openshift/openshift-ansible/issues/9753, here we don't specify `openshift_release`, so `docker_image_availability` passes.

Internal reproducer J:2718064.

**Comment by michaelgugino** [comment]
@adelton openshift_node_labels has no effect in 3.10+

Please create a pastebin with -vvv output.

**Comment by adelton** [comment]
The `openshift_node_labels` are there because I use the same tests for different OpenShift versions. My understanding is that it should not harm.

The verbose output of

```
ansible-playbook -vvv -i ansible-inventory /mnt/tests/OpenShift/Origin/install/dist/openshift-ansible/playbooks/deploy_cluster.yml
```

is at https://paste.fedoraproject.org/paste/IV3hlCiaqitGyTZYJccZYg.

Internal reproducer J:2735947.

**Comment by michaelgugino** [comment]
Looks like epel atomic package is broken:

'Error: Package: 1:atomic-1.22.1-22.git5a342e3.el7.x86_64 (rhel-7-server-extras-rpms)\n           Requires: python-docker-py >= 1.7.2-1\n '
...
'nPackage python-docker-py-1.10.6-4.el7.noarch is obsoleted by python2-docker-3.3.0-1.el7.noarch which is already installed'

@sdodson @vrutkovs I have seen one of you two do something with this package recently.  Anything we can do on our end?

**Comment by jfchevrette** [comment]
Getting the same on CentOS 7.5.1804 /w extras and centos-openshift-origin310 repos. (stable or -testing result in the same error). I do not use EPEL.

Is `atomic` a new requirement? My cluster seem to install and run just fine if I comment out the package in `default_r_openshift_node_image_prep_packages`. Looks like it was added in #8953 

```
Error: Package: 1:atomic-1.22.1-22.git5a342e3.el7.x86_64 (extras)
           Requires: python-docker-py >= 1.7.2-1
           Available: docker-python-1.4.0-115.el7.x86_64 (extras)
               python-docker-py = 1.4.0-115.el7
           Available: python-docker-py-1.10.6-3.el7.noarch (extras)
               python-docker-py = 1.10.6-3.el7
           Available: python-docker-py-1.10.6-4.el7.noarch (extras)
               python-docker-py = 1.10.6-4.el7
           Available: python2-docker-2.4.2-1.2.el7.noarch (centos-openshift-origin310-testing)
               python-docker-py = 2.4.2-1.2.el7
           Installed: python2-docker-3.3.0-1.el7.noarch (@centos-openshift-origin-testing)
               Not found
```

**Comment by MartijnStraatman** [comment]
Same here on CentOS 7.5.1805.

**Comment by DanyC97** [comment]
@jfchevrette @MartijnStraatman any chance you can  pls give it a go using -[candidate](https://cbs.centos.org/repos/paas7-openshift-origin310-candidate/x86_64/os/Packages/) repo  ? 

last time when @vrutkovs had some issues i worked with him and enabled some packages there however i haven't promoted the pkgs to -testing & -release repos .

@sdodson shall i go and tag them ?

**Comment by MartijnStraatman** [comment]
How do I use -candidate?

Op wo 5 sep. 2018 om 22:06 schreef Dani Comnea <notifications@github.com>

**Comment by adelton** [comment]
@DanyC97, what's the recommended way to get `ansible` for that https://cbs.centos.org/repos/paas7-openshift-origin310-candidate/x86_64/os/Packages/ repo?
On CentOS 7, I get
```
Error: Package: openshift-ansible-3.10.51-1.git.0.44a646c.el7.noarch (cbs.centos.org_repos_paas7-openshift-origin310-candidate_x86_64_os_)
           Requires: ansible >= 2.4.3
           Available: ansible-2.4.2.0-2.el7.noarch (extras)
               ansible = 2.4.2.0-2.el7
```

**Comment by adelton** [comment]
And on RHEL 7.5, I can get `ansible-2.6.5-1.el7ae` from `rhel-7-server-ansible-2.6-rpms` but the `python2-docker` situation is still the same.
Could you share details of your inventory file which can be used to install the latest OpenShift Origin?

**Comment by DanyC97** [comment]
> @DanyC97, what's the recommended way to get `ansible` for that https://cbs.centos.org/repos/paas7-openshift-origin310-candidate/x86_64/os/Packages/ repo?
> On CentOS 7, I get
> 
> ```
> Error: Package: openshift-ansible-3.10.51-1.git.0.44a646c.el7.noarch (cbs.centos.org_repos_paas7-openshift-origin310-candidate_x86_64_os_)
>            Requires: ansible >= 2.4.3
>            Available: ansible-2.4.2.0-2.el7.noarch (extras)
>                ansible = 2.4.2.0-2.el7
> ```

i'm working with CentOS infra guys to build the ansible rpm and and then tag it into each paas7-openshift-origin* repo to avoid having to bounce to epel and or rhel. In this way we will also have better control/ alignment with the ansible version mentioned in the spec file of each release.

currently every user is ignoring the notes o the ansible version supported and then we end up with issues poping everywhere ...this must stop
